### PR TITLE
Fixes #17047: Load all file from stdlib automatically

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/17047

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
@peckpeck I get random test failures, I gues it(s because files need to get loaded in a precise order.

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
Doesn't seem linked to load order actually, I also get these on current master.

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
Ready for review.

File: rudder-lang/src/compile.rs [hunk 1]
```diff
@@ -8,29 +8,52 @@ use crate::{
...
-    let stdlib = libs_dir.join("stdlib.rl");
-    parse_file(past, sources, &stdlib, "stdlib.rl")?;
+    let walker = WalkDir::new(libs_dir)
+        .into_iter()
+        .filter(|r| r.as_ref().map(|e| is_rl_file(e.path())).unwrap_or(true));
+    for entry in walker {
+        match entry {
+            Ok(e) => {
+                let path = e.path();
+                let filename = sources.alloc(
```
Comment (peckpeck) [inline rudder-lang/src/compile.rs:34]: this part could be in parse_file so that we can remove one parameter from it

Comment (amousset) [reply rudder-lang/src/compile.rs:34]: indeed, I wanted the keep the previous behavior of only passing the leaf file name for stdlib files, but if it does not matter it's indeed easier.

Comment (peckpeck) [reply rudder-lang/src/compile.rs:34]: the previous behavior is "historical" in the sense that is was mandatory before putting them in the Arena to have the proper lifetime for the file name

Comment (amousset) [reply rudder-lang/src/compile.rs:34]: :+1:

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**